app/main.py

```diff
@@ -1,3 +1,68 @@
+from typing import Any
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Any) -> "Distance":
+        if isinstance(other, Distance):
+            return Distance(km=self.km + other.km)
+        elif isinstance(other, (int, float)):
+            return Distance(km=self.km + other)
+
+    def __iadd__(self, other: Any) -> "Distance":
+        if isinstance(other, Distance):
+            self.km += other.km
+        elif isinstance(other, (int, float)):
+            self.km += other
+        return self
+
+    def __mul__(self, other: Any) -> "Distance":
+        if isinstance(other, (int, float)):
+            return Distance(km=self.km * other)
+        elif isinstance(other, Distance):
+            raise TypeError("Cannot multiply two Distance instances")
+        raise TypeError("Division is only supported with int or float")
+
+    def __truediv__(self, other: Any) -> "Distance":
```

Comment:
Ensure that the `other` parameter in the `__truediv__` method is a number (either `int` or `float`) before performing the division. This will prevent potential runtime errors.